c++/src/kj/compat/tls.c++

```diff
@@ -183,22 +183,22 @@ public:
 
   kj::Promise<void> connect(kj::StringPtr expectedServerHostname) {
     if (!SSL_set_tlsext_host_name(ssl, expectedServerHostname.cStr())) {
-      return getOpensslError();
+      throw getOpensslError();
     }
 
     X509_VERIFY_PARAM* verify = SSL_get0_param(ssl);
     if (verify == nullptr) {
-      return getOpensslError();
+      throw getOpensslError();
     }
 
     if (isIpAddress(expectedServerHostname)) {
       if (X509_VERIFY_PARAM_set1_ip_asc(verify, expectedServerHostname.cStr()) <= 0) {
-        return getOpensslError();
+        throw getOpensslError();
       }
     } else {
       if (X509_VERIFY_PARAM_set1_host(
           verify, expectedServerHostname.cStr(), expectedServerHostname.size()) <= 0) {
-        return getOpensslError();
+        throw getOpensslError();
```

Comment:
I wonder if maybe these bare `throw`s should be `kj::throwFatalException(getOpensslError())` instead?  Or `throwOpensslError()`, which seems to be the same thing.

Reply:
Yes, we should not have bare `throw`s ever.